src/dos/dos_files.cpp

```diff
@@ -1856,3 +1851,16 @@ bool DOS_UnlockFile(const uint16_t entry, const uint32_t pos, const uint32_t len
 	DOS_SetError(DOSERR_LOCK_VIOLATION);
 	return false;
 }
+
+void DOS_ClearDrivesAndFiles()
+{
+	// Clear all the DOS files. This calls the files' derived destructors,
+	// which might be LocalFiles, OverlayFile, etc.
+	for (auto& f : Files) {
+		f.reset();
+	}
+
```

Comment:
@weirddan455, I wanted to use the `.fill(nullptr)` like you did for the Drives array, but I hit some issues with the `operator =` being ambiguous for the DOS_File class, so had to switch to this looping approach.

Reply:
I'm getting the same error.  I think it's because the `DOS_Drive` array contains `shared_ptr` while the `DOS_File` array contains `unique_ptr`.  `.fill()` doesn't have an overload for r-value references and `unique_ptr` can only be moved, never copied.

Although if you put `f = nullptr` in the loop it works fine.  Just weird C++ things :roll_eyes:  `.fill()` must be constructing a temporary and then attempting to do a copy, which fails since that operator is deleted for `unique_ptr`.

Reply:
I guess it makes sense in a way because you can't move a single object to 256 different places so it has to copy.

Reply:
Thanks @weirddan455, that makes sense. 

Replaced the reset() with the = nullptr; thanks!

Reply:
`f.reset()` is probably more idiomatic for smart pointers.  I was just pointing out assigning nullptr in a loop works because I thought it was interesting.  Doesn't really matter either way though, it does the same thing.

Reply:
Good point; I probably should have stuck with it, but figured pairing it in style w/ the Drives was a better fit. Agree though it's not a big deal though :-), but will remember for next time.